test/e2e/framework/networking_utils.go

```diff
@@ -186,13 +186,12 @@ func (config *NetworkingTestConfig) DialFromContainer(protocol, containerIP, tar
 			if err := json.Unmarshal([]byte(stdout), &output); err != nil {
 				Logf("WARNING: Failed to unmarshal curl response. Cmd %v run in %v, output: %s, err: %v",
 					cmd, config.HostTestContainerPod.Name, stdout, err)
-				continue
-			}
-
-			for _, hostName := range output["responses"] {
-				trimmed := strings.TrimSpace(hostName)
-				if trimmed != "" {
-					eps.Insert(trimmed)
+			} else {
+				for _, hostName := range output["responses"] {
+					trimmed := strings.TrimSpace(hostName)
+					if trimmed != "" {
+						eps.Insert(trimmed)
+					}
```

Comment:
> switch the service from nodePort -> normal clusterIP

The only similar e2e test I could remember is https://github.com/kubernetes/kubernetes/blob/master/test/e2e/service.go#L773-L802. But it is using a different function.

And for https://github.com/kubernetes/kubernetes/blob/master/test/e2e/networking.go#L174, it is also calling a different function [`DialFromNode()`](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/networking_utils.go#L213-L264), in which it just trims the stdout without calling Unmarshal(). I think that function would not be affected by this change?